src/Square3x3.java

```diff
@@ -0,0 +1,206 @@
+/**
+ * Maman 13 - Class Square3x3
+ *
+ * @author Noa Amir
+ * @version 2022a
+ */
+
+public class Square3x3 {
+
+    // initialize final variables
+    private final int SIZE = 3;
+    private final int DEFAULT_CELL_VALUE = -1;
+    private final boolean[] WHICH_NUMBERS_APPEAR = new boolean[SIZE*SIZE+1];
+
+    // instance variables
+    private int[][] square;
+
+    // constructors
+
+    /**
+     * Construct a new Square3x3. Initialize all cells value to -1
+     */
+    public Square3x3() {
+        this.initSquare(DEFAULT_CELL_VALUE);
+    }
+
+    /**
+     * Construct a new Square3x3. Copy values from a given array to Square3x3 cells
+     *
+     * @param array - array to copy the values from
+     */
+    Square3x3(int[][] array) {
+        this.initSquare(DEFAULT_CELL_VALUE);
+
+        int rows = Math.min(array.length, SIZE);
+
+        for (int i = 0; i < rows; i++) {
+            int cols = Math.min(array[i].length, SIZE);
+            for (int j = 0; j < cols; j++) {
+                square[i][j] = array[i][j];
+            }
+        }
+    }
+
+    /**
+     * Construct a new Square3x3 using another Square3x3
+     *
+     * @param other - The Square3x3 from which to construct the new object
+     */
+    public Square3x3(Square3x3 other) {
+        if (other != null){
+            square = new int[SIZE][SIZE];
+            for (int i = 0; i < SIZE; i++) {
+                for (int j = 0; j < SIZE; j++) {
+                    this.setXY(i, j, other.getCell(i, j));
+                }
+            }
+        }
+        else
+            this.initSquare(DEFAULT_CELL_VALUE);
+    }
+
+    // getters & setters
+
+    /**
+     * Returns the value in [row][col] cell of the Square3x3 object
+     *
+     * @return value in [row][col] cell
+     */
+    public int getCell(int row, int col) {
+        if (validIndex(row) && validIndex(col))
+            return square[row][col];
+        return -1;
```

Comment:
-1 --> should be a final (a new one, specifying an invalid value).

Also, in the description please explicitly describe what value is expected to return for an invalid index.